### PR TITLE
add force start command and add Veridad Pass

### PR DESCRIPTION
**Pull request by Varunda**
full changes:

add Veridad Pass to the map pool.

add a `=fstart` command for admins to start lobbies that have less than the number of players needed to start.

fix crash on lobby start if the Mongo collection `restart_data` is empty.

fix 2 player lobbies not starting due to the player picker not showing any players (instead just goes to faction select).

fix lobbies not starting if no jaeger calendar ID is given.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,4 @@ doc/build/
 .pdm.toml
 __pypackages__
 .pdm-python
+pdm.py
```
Comment (yakMM) [inline .gitignore:23]: Why is this needed?

File: bot/cogs/admin.py [hunk 2]
```diff
@@ -257,6 +257,26 @@ async def pog(self, ctx, *args):
             return
         await disp.WRONG_USAGE.send(ctx, ctx.command.name)
 
+    @commands.command()
+    @commands.guild_only()
+    async def fstart(self, ctx, *args):
```
Comment (yakMM) [inline bot/cogs/admin.py:262]: Let's just go with `=start` no?

File: bot/cogs/admin.py [hunk 3]
```diff
@@ -257,6 +257,26 @@ async def pog(self, ctx, *args):
...
+        if ctx.channel.id != cfg.channels["lobby"]: # only allowed in lobby channel
+            await disp.WRONG_CHANNEL_2.send(ctx, ctx.command.name, f"<#{ctx.channel.id}>")
+            return
+
+        if (lobby.get_lobby_len() == 0): # don't allow starting with 0 players
+            await disp.LB_FORCE_START_ZERO_PLAYERS.send(ctx)
+            return
+
+        # TODO: is this check needed? can you successfully have a 5v6 lobby?
```
Comment (yakMM) [inline bot/cogs/admin.py:271]: Yeah I think it's fine, otherwise people will have to bench each other anyways

File: bot/match/classes/match.py [hunk 4]
```diff
@@ -134,6 +134,9 @@ def change_check(self, arg):
     def spin_up(self, p_list):
         if not self.__objects:
             raise AttributeError("Match instance is not bound, no attribute 'spin_up'")
+        if (Match._last_match_id == None): # if there was no last match, make sure it exists
```
Comment (yakMM) [inline bot/match/classes/match.py]:
```suggestion
        if Match._last_match_id is None: # if there was no last match, make sure it exists
```

File: bot/cogs/admin.py [hunk 5]
```diff
@@ -257,6 +257,26 @@ async def pog(self, ctx, *args):
...
         await disp.WRONG_USAGE.send(ctx, ctx.command.name)
 
+    @commands.command()
+    @commands.guild_only()
+    async def fstart(self, ctx, *args):
+        if ctx.channel.id != cfg.channels["lobby"]: # only allowed in lobby channel
+            await disp.WRONG_CHANNEL_2.send(ctx, ctx.command.name, f"<#{ctx.channel.id}>")
+            return
+
+        if (lobby.get_lobby_len() == 0): # don't allow starting with 0 players
```
Comment (yakMM) [inline bot/cogs/admin.py]:
```suggestion
        if lobby.get_lobby_len() == 0: # don't allow starting with 0 players
```

File: bot/cogs/admin.py [hunk 6]
```diff
@@ -257,6 +257,26 @@ async def pog(self, ctx, *args):
...
+            await disp.LB_FORCE_START_ZERO_PLAYERS.send(ctx)
+            return
+
+        # TODO: is this check needed? can you successfully have a 5v6 lobby?
+        if (lobby.get_lobby_len() % 2 != 0):
```
Comment (yakMM) [inline bot/cogs/admin.py]:
```suggestion
        if lobby.get_lobby_len() % 2 != 0:
```

File: bot/match/processes/player_picking.py [hunk 7]
```diff
@@ -50,10 +52,15 @@ async def init(self):
         """ Init the match channel, ping players, find two captains \
             and ask them to start picking players.
         """
-        # Ready for players to pick
-        ctx = self.interaction_handler.get_new_context(self.match.channel)
-        await disp.MATCH_SHOW_PICKS.send(ctx, self.match.teams[0].captain.mention,
-                                         match=self.match.proxy)
+
+        if (len(self.players) == 0):
```
Comment (yakMM) [inline bot/match/processes/player_picking.py]:
```suggestion
        if not self.players:
```

File: bot/match/processes/player_picking.py [hunk 8]
```diff
@@ -50,10 +52,15 @@ async def init(self):
...
+            log.debug(f"there are 0 players, meaning this is a 2 person lobby with 2 captains already!")
```
Comment (yakMM) [inline bot/match/processes/player_picking.py:57]: Shouldn't we just enforce 4 min players?

File: bot/modules/jaeger_calendar.py [hunk 9]
```diff
@@ -22,6 +22,11 @@ def init(secret_file):
 def get_booked_bases(base_class, booked_bases_list):  # runs on class init, saves a list of booked bases at the time of init to self.booked
     index_start = index_end = None
     gc = service_account(filename=_secret_file)
+
+    if (cfg.database["jaeger_cal"] is None or cfg.database["jaeger_cal"] == ""):
```
Comment (yakMM) [inline bot/modules/jaeger_calendar.py]:
```suggestion
    if cfg.database["jaeger_cal"] is None or cfg.database["jaeger_cal"] == "":
```

File: bot/modules/jaeger_calendar.py [hunk 10]
```diff
@@ -22,6 +22,11 @@ def init(secret_file):
...
+        log.info("config value 'jaeger_cal' is not set, skipping booked bases check")
```
Comment (yakMM) [inline bot/modules/jaeger_calendar.py]:
```suggestion
        log.warning("config value 'jaeger_cal' is not set, skipping booked bases check")
```

**Review by yakMM — COMMENTED** [review]
Thank you for this!

File: .gitignore (hunk 1 above)
Comment (Varunda) [reply .gitignore:23]: was an extra file in my repo when i went to commit, not sure where it came from

File: bot/cogs/admin.py (hunk 2 above)
Comment (Varunda) [reply bot/cogs/admin.py:262]: i have a slight preference to imply that this is an admin forcing a start, but don't mind either way realy

File: bot/match/processes/player_picking.py (hunk 8 above)
Comment (Varunda) [reply bot/match/processes/player_picking.py:57]: being able to test with 2 accounts was really nice. i don't see a reason to not use this

File: bot/modules/jaeger_calendar.py (hunk 10 above)
Comment (Varunda) [reply bot/modules/jaeger_calendar.py]: not sure this is a warning. it's perfectly valid for the jaeger calendar value to not be set

File: bot/cogs/admin.py (hunk 2 above)
Comment (yakMM) [reply bot/cogs/admin.py:262]: Ok that's fine

File: bot/modules/jaeger_calendar.py [hunk 11]
```diff
@@ -22,6 +22,11 @@ def init(secret_file):
 def get_booked_bases(base_class, booked_bases_list):  # runs on class init, saves a list of booked bases at the time of init to self.booked
     index_start = index_end = None
     gc = service_account(filename=_secret_file)
+
+    if cfg.database["jaeger_cal"] is None or cfg.database["jaeger_cal"] == "":
```
Comment (yakMM) [inline bot/modules/jaeger_calendar.py]:
```suggestion
    if not cfg.database["jaeger_cal"]:
```
This covers both cases

File: bot/modules/jaeger_calendar.py (hunk 10 above)
Comment (yakMM) [reply bot/modules/jaeger_calendar.py]: Nah it should be warning IMO, it's one of the feature of the bot, having at least a warning might prevent silently forgetting to fill the field in prod.

File: bot/match/processes/player_picking.py (hunk 8 above)
Comment (yakMM) [reply bot/match/processes/player_picking.py:57]: Are you sure it's not breaking something else?

Comment (Varunda) [reply bot/match/processes/player_picking.py:57]: i successfully ran both a 2 player, and a 4 player (force started) lobby. if you really are worried about this, we can remove it, but it makes testing harder
